src/main/java/pl/mmieczak/controller/MainController.java

```diff
@@ -0,0 +1,86 @@
+package pl.mmieczak.controller;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import pl.mmieczak.model.Category;
+import pl.mmieczak.model.Product;
+import pl.mmieczak.model.ProductRepository;
+import pl.mmieczak.utils.Utilities;
+
+import java.math.BigDecimal;
+import java.util.List;
+
+@Controller
+public class MainController {
+
+    ProductRepository productRepository;
+
+    public MainController(ProductRepository productRepository) {
```

Comment:
`@Autowired` - generalnie działa to na konstruktorze bez `@Autowired` (ot taki mechanizm od springa) ale zawsze staraj się to tutaj dodawać.

Reply:
Z tym Autowired Marcin ostatnio nam wytłumaczył, że poprzez konstruktor jest to bardziej elegancka metoda i ten @Autowired jest tam odpalany tak czy inaczej, i nie musimy tego robić. Więc które podejscie jednak jest bardziej “prawilne” :)

Reply:
Odpisałem na slacku :)